### PR TITLE
feat: custom editor/renderer spec changes

### PR DESCRIPTION
**Pull request by dongwoo-kim**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
- editor / editorOptions -> editor.type / editor.options
- renderer / rendererOptions -> renderer.type / renderer.options

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/store/column.ts [hunk 1]
```diff
@@ -24,15 +39,39 @@ const defMinWidth = {
...
+function getEditorOptions(editor?: OptCellEditor): CellEditorOptions | null {
+  if (isFunction(editor)) {
+    return { type: editor };
+  }
+  if (isString(editor)) {
+    return getBuiltInEditorOptions(editor);
   }
-  return { editor, editorOptions };
+  if (isObject(editor)) {
+    if (isString(editor.type)) {
```
Comment (seonim-ryu) [inline src/store/column.ts]: if문 묶는건 어떤가요??
```js
if (isObject(editor) && isString(editor.type)) { ... }
```

File: src/store/column.ts [hunk 2]
```diff
@@ -24,15 +39,39 @@ const defMinWidth = {
...
+      return getBuiltInEditorOptions(editor.type, editor.options);
+    }
+    return editor as CellEditorOptions;
+  }
```
Comment (js87zz) [inline src/store/column.ts]: `isString(editor.type) ? getBuiltInEditorOptions(editor.type, editor.options) : editor as CellEditorOptions;`로 간소화하는게 좋을 듯 합니다.

File: src/view/editingLayerInner.tsx [hunk 3]
```diff
@@ -71,27 +71,27 @@ export class EditingLayerInnerComp extends Component<Props> {
           dispatch('sort', columnName, sortOptions.ascending);
         }
       }
-      if (typeof this.editor.finish === 'function') {
-        this.editor.finish();
+      if (typeof this.editor.beforeDestroy === 'function') {
```
Comment (js87zz) [inline src/view/editingLayerInner.tsx]: `if (isFunction(this.editor.beforeDestroy)) {...`로 사용 가능합니다!

File: src/store/column.ts [hunk 4]
```diff
@@ -89,31 +128,33 @@ export function createColumn(
...
     copyOptions: { ...gridCopyOptions, ...copyOptions },
     baseWidth: (width === 'auto' ? 0 : width) || 0,
     minWidth: minWidth || columnOptions.minWidth || defMinWidth.COLUMN, // @TODO meta tag 체크 여부
     relationMap: getRelationMap(relations || []),
     related: includes(relationColumns, name),
     sortable,
-    ...getEditorInfo(editor, editorOptions),
     validation: validation ? { ...validation } : {},
+    renderer: rendererOptions,
+    ...(editorOptions ? { editor: editorOptions } : null),
```
Comment (seonim-ryu) [inline src/store/column.ts]: 스프레드 사용할 때 `null` 사용 강제할까요ㅎ
```js
...(editorOptions && { editor: editorOptions })
```

File: src/view/editingLayerInner.tsx (hunk 3 above)
Comment (seonim-ryu) [inline src/view/editingLayerInner.tsx]: common 헬퍼의 `isFunction` 사용할 수 있겠네요ㅎ

**Comment by js87zz** [comment]
리뷰 완료입니다. 스펙 정리하느라 고생하셨습니다. 👍

**Comment by seonim-ryu** [comment]
[6/11] 리뷰 완료하였습니다! 달리느라 고생 많으셨어요 ;ㅅ;

File: src/store/column.ts (hunk 1 above)
Comment (dongwoo-kim) [reply src/store/column.ts]: 이건 밑에 return 문 때문에 묶을 수가 없을 것 같아요.

File: src/store/column.ts (hunk 2 above)
Comment (dongwoo-kim) [reply src/store/column.ts]: 넵~

File: src/store/column.ts (hunk 4 above)
Comment (dongwoo-kim) [reply src/store/column.ts]: 음..`&&` 가 더 깔끔하겠네요 ㅎ

File: src/view/editingLayerInner.tsx (hunk 3 above)
Comment (dongwoo-kim) [reply src/view/editingLayerInner.tsx]: ㅎㅎ 넵

**Pull request merged**